client/src/main/java/org/eclipse/hono/client/device/amqp/AmqpAdapterClientFactory.java

```diff
@@ -33,11 +34,12 @@
      *
      * @param connection The connection to use.
      * @param tenantId The ID of the tenant for which the connection is authenticated.
+     * @param samplerFactory The sampler factory to use.
      * @return The factory.
      * @throws NullPointerException if any of the parameters is {@code null}
      */
-    static AmqpAdapterClientFactory create(final HonoConnection connection, final String tenantId) {
-        return new AmqpAdapterClientFactoryImpl(connection, tenantId);
+    static AmqpAdapterClientFactory create(final HonoConnection connection, final String tenantId, final SendMessageSampler.Factory samplerFactory) {
```

Comment:
IMHO we should also provide a factory method that doesn't require a sampler factory because AMQP adapter clients will usually be external to a Hono instance and thus most likely not integrate with Hono's metrics backend. WDYT?